pandas/tests/reshape/merge/test_merge_anti.py

```diff
@@ -0,0 +1,502 @@
+                    }
+                ).astype({"B_x": object}),
+                "anti_right",
+                None,
+                ["A"],
+                ["C"],
+            ),
+        ],
+    )
+    def test_anti_with_nan(self, expected, how, on, left_on, right_on):
+        left = DataFrame({"A": [np.nan, 2, np.nan], "B": ["a", 2, 3]})
+        right = DataFrame({"C": [1, 3, np.nan], "B": ["a", 2, "c"]})
+        result = merge(left, right, on=on, how=how, left_on=left_on, right_on=right_on)
+        tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "expected, how, left_on, right_on",
+        [
+            (
+                DataFrame({"A": [np.nan, pd.NA], "B": ["a", 3], "C": [np.nan, np.nan]}),
+                "anti_left",
+                "B",
+                "B",
+            ),
+            (
+                DataFrame(
+                    {"A": [np.nan, np.nan], "B": [pd.NA, "c"], "C": [1, np.nan]}
+                ).astype({"A": object}),
+                "anti_right",
+                "B",
+                "B",
+            ),
+            (
+                DataFrame(
+                    {
+                        "A": [np.nan, pd.NA, np.nan, np.nan],
+                        "B": ["a", 3, pd.NA, "c"],
+                        "C": [np.nan, np.nan, 1, np.nan],
+                    }
+                ),
+                "anti_full",
+                "B",
+                "B",
+            ),
+            (
+                DataFrame(
+                    {
+                        "A": [2, pd.NA],
+                        "B_x": [2, 3],
+                        "C": [np.nan, np.nan],
+                        "B_y": [np.nan, np.nan],
+                    }
+                ).astype({"B_x": object, "B_y": object}),
+                "anti_left",
+                "A",
+                "C",
+            ),
+            (
+                DataFrame(
+                    {
+                        "A": [np.nan, np.nan],
+                        "B_x": [np.nan, np.nan],
+                        "C": [1.0, 3],
+                        "B_y": [pd.NA, 2],
+                    }
+                ).astype(
+                    {
+                        "A": object,
+                        "B_x": object,
+                    }
+                ),
+                "anti_right",
+                "A",
+                "C",
+            ),
+            (
+                DataFrame(
+                    {
+                        "A": [2, pd.NA, np.nan, np.nan],
+                        "B_x": [2, 3, np.nan, np.nan],
+                        "C": [np.nan, np.nan, 1, 3],
+                        "B_y": [np.nan, np.nan, pd.NA, 2],
+                    }
+                ).astype({"B_x": object}),
+                "anti_full",
+                "A",
+                "C",
+            ),
+        ],
+    )
+    def test_anti_with_nan_and_NA(self, expected, how, left_on, right_on):
```

Comment:
Is this test purely testing `pd.NA`? 

In this case it seems overkill. Logically speaking if the tests above imply everything is working for `nan`, and a simple test shows `pd.Na` works equivalently to `nan` then all tests above should work for `pd.NA` so we don't have to repeat everything again.

Reply:
actually this was to check `np.nan` is not matched with `pd.NA`